console/src/components/Services/Data/utils.js

```diff
@@ -784,3 +786,40 @@ WHERE
 
 export const isColTypeString = colType =>
   ['text', 'varchar', 'char', 'bpchar', 'name'].includes(colType);
+
+export const cascadeUpQueries = (upQueries = []) =>
+  upQueries.map((i = {}) => {
+    if (i.type === 'run_sql' || i.type === 'untrack_table') {
+      return {
+        ...i,
+        args: {
+          ...i.args,
+          cascade: true,
+        },
+      };
+    }
+    return i;
+  });
+
+export const getRetryStatus = (isRetry, err = {}) => {
```

Comment:
It got a bit more complicated. You return `errorObj` and it's only used when `canRetry` is true, but you're returning it in all paths here. I'd move `isRetry` outside of this function because it's not anywhere except fro early return. 

What about something like this?

```js
const makeMigrationCall(...) => {
 // ...


  const onError = err => {
      if (!isRetry) {
        const dependecyError = getRetryStatus(isRetry, err);
        if (dependecyError) return retryMigration(dependecyError, errorMsg);
      }  

     dispatch(handleMigrationErrors(errorMsg, err));
     customOnError(err);
  }
}

export const getDependencyError = (isRetry, err = {}) => {
  if (err.code == ERROR_CODES.dependencyError.code) return err;
  
  if (err.code == ERROR_CODES.dataApiError.code) {
      const actualError = isJsonString(err.message) ? JSON.parse(err.message) : {};
      if (actualError.code == ERROR_CODES.dependencyError.code) return { ...actualError, message: actualError.error }
  }
  return null;
}
```

Reply:
Done